pkg/imageset/updateinstaller.go

```diff
@@ -137,91 +120,71 @@ func (o *UpdateInstallerImageOptions) Validate() error {
-	successFileName := path.Join(o.WorkDir, "success")
-	success := false
-	o.log.Debug("starting to wait for success result")
-	err := wait.PollImmediate(pollInterval, releaseInfoCommandTimeout, func() (bool, error) {
-		_, err := os.Stat(successFileName)
-		if err != nil {
-			if !os.IsNotExist(err) {
-				o.log.WithError(err).Warningf("unexpected error trying to stat %s", successFileName)
-			}
-			return false, nil
-		}
-		result, err := ioutil.ReadFile(successFileName)
-		if err != nil {
-			o.log.WithError(err).Warningf("unexpected error trying to read %s", successFileName)
-			return false, nil
-		}
-		if len(string(result)) == 0 {
-			// keep waiting for content
-			return false, nil
-		}
-		o.log.Debugf("contents of success file: %s", strings.TrimSpace(string(result)))
-		success = strings.TrimSpace(string(result)) == "1"
-		return true, nil
-	})
+// Run updates the given ClusterDeployment based on the image-references file.
+func (o *UpdateInstallerImageOptions) Run() (returnErr error) {
+	cd := &hivev1.ClusterDeployment{}
+	cdName := types.NamespacedName{Namespace: o.ClusterDeploymentNamespace, Name: o.ClusterDeploymentName}
+	logger := o.log.WithField("clusterdeployment", cdName)
+	logger.Debug("fetching clusterdeployment")
+	err := o.client.Get(context.TODO(), cdName, cd)
 	if err != nil {
-		o.log.WithError(err).Error("timed out waiting for result")
-		return err
+		return errors.Wrap(err, "failed to get ClusterDeployment")
 	}
 
-	if success {
-		o.log.Debugf("the oc release info command was successful")
-		installerImageFileName := path.Join(o.WorkDir, "installer-image.txt")
-		installerImageBytes, err := ioutil.ReadFile(installerImageFileName)
-		if err != nil {
-			o.log.WithError(err).Error("could not read install image file")
-			return err
+	defer func() {
+		if returnErr == nil {
+			return
 		}
-		installerImage := strings.TrimSpace(string(installerImageBytes))
-		o.log.Debugf("contents of installer-image.txt: %s", installerImage)
+		o.setImageResolutionErrorCondition(cd, returnErr)
+	}()
 
-		cliImageFileName := path.Join(o.WorkDir, "cli-image.txt")
-		cliImageBytes, err := ioutil.ReadFile(cliImageFileName)
-		if err != nil {
-			o.log.WithError(err).Error("could not read install image file")
-			return err
-		}
-		cliImage := strings.TrimSpace(string(cliImageBytes))
-		o.log.Debugf("contents of cli-image.txt: %s", cliImage)
-
-		return o.updateInstallerImage(installerImage, cliImage)
+	imageStreamData, err := ioutil.ReadFile(filepath.Join(o.WorkDir, imageReferencesFilename))
+	if err != nil {
+		return errors.Wrapf(err, "could not read %s file", imageReferencesFilename)
+	}
+	is := &imageapi.ImageStream{}
+	if err := yaml.Unmarshal(imageStreamData, &is); err != nil {
+		return errors.Wrap(err, "unable to load release image-references")
+	}
+	if is.Kind != "ImageStream" || is.APIVersion != "image.openshift.io/v1" {
+		return errors.Wrap(err, "unrecognized image-references in release payload")
```

Comment:
Can you confirm this works as far back as 4.1? We could have the discussion if not but it's a lot easier if we can just keep supporting 4.1.

Reply:
Yep. I will check on that.

Reply:
This worked with https://registry.svc.ci.openshift.org/v2/ocp/release/manifests/4.1.0-0.nightly-2020-03-21-052817. Testing did reveal a problem where the pull secret was not being used. I updated the code to add the pull secret to the pod spec.